cwms-data-api/src/test/java/cwms/cda/api/OpenApiDocTest.java

```diff
@@ -0,0 +1,486 @@
+        assertAll(buildTestAssertions(compilationUnit, testInfo));
+    }
+
+    @MethodSource(value = "getCrudHandlerDocInfo")
+    @ParameterizedTest
+    void test_crud_handler_documentation(OpenApiDocTestInfo testInfo) throws IOException {
+        CompilationUnit compilationUnit = OpenApiTestHelper.readCompilationUnit(testInfo.getClazz());
+        assertAll(buildTestAssertions(compilationUnit, testInfo));
+    }
+
+    @Test
+    void test_time_series_controller() throws IOException {
+        OpenApiDocTestInfo testInfo = OpenApiTestHelper.readOpenApiDocs(CrudHandler.class, TimeSeriesController.class);
+        CompilationUnit compilationUnit = OpenApiTestHelper.readCompilationUnit(testInfo.getClazz());
+        assertAll(buildTestAssertions(compilationUnit, testInfo));
+    }
+
+    private Stream<Executable> buildTestAssertions(CompilationUnit compilationUnit, OpenApiDocTestInfo testInfo) {
+        return testInfo.getMethodDocs()
+                       .stream()
+                       .map(docInfo -> validateOpenApiDoc(compilationUnit, docInfo, testInfo.getClazz()));
+    }
+
+    private Executable validateOpenApiDoc(CompilationUnit unit, OpenApiDocInfo testInfo, Class<?> clazz){
+        Executable output;
+        if (testInfo.isIgnored()) {
+            output = testIgnoredMethod(unit, testInfo, clazz);
+        } else {
+            OpenApiParamUsage parsedParamInfo = parseParamInfo(unit, clazz, testInfo.getMethod());
+            output = testMethod(testInfo, parsedParamInfo);
+        }
+        return output;
+    }
+
+    private Executable testIgnoredMethod(CompilationUnit unit, OpenApiDocInfo testInfo, Class<?> clazz) {
+
+        // 16/24 cases throw an UnsupportedOperationException, so verify that behavior here for now.
+        // Other cases use the context to provide additional feedback, but the feedback is not consistent.
+        // Could be an improvement for the corps to define it more strictly here.
+        MethodDeclaration method = getMethodDeclaration(unit, testInfo.getMethod());
+        boolean throwsException = method.findAll(ThrowStmt.class)
+                                        .stream()
+                                        .anyMatch(throwStmt -> {
+                                            if (throwStmt.getExpression() instanceof ObjectCreationExpr) {
+                                                ObjectCreationExpr creation = (ObjectCreationExpr) throwStmt.getExpression();
+                                                return creation.getType().getNameAsString().equals("UnsupportedOperationException");
+                                            }
+                                            return false;
+                                        });
+        return () -> assertTrue(throwsException, clazz.getSimpleName() + "::" + testInfo.getMethod().getName() + " is marked as ignored, but does not throw an UnsupportedOperationException.");
+    }
```

Comment:
I'm not convinced that we need to look for UnsupportedOperationException.  Or, I'm not convinced that is the only "right" way for something to be unimplemented.    There's at least one controller that does: 
```
 ctx.status(HttpCode.NOT_IMPLEMENTED).json(CdaError.notImplemented());
```
And that seems pretty clear and clean to me.  I'd be curious what other patterns are in use besides those two.

Reply:
One could argue, in light of that, that we should determine a single right way for us.